packages/core/src/App/Components/Elements/LiveChat/use-livechat.ts

```diff
@@ -101,6 +101,11 @@ const useLiveChat = (has_cookie_account = false) => {
         });
     };
 
+    useEffect(() => {
+        onHistoryChange();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [window.location.search, onHistoryChange]);
```

Comment:
@sergei-deriv Hmmm I'm still unsure about this newly added `useEffect`, The dependencies and the body of the `useEffect` don’t match, and `window.location.search` has not been used anywhere so why do we need to do something on this side effect? 🤔

Reply:
@farzin-deriv We need this because live-chat must be reinitialised every time when an user changes query parameter. Earlier it reinitialised only when uri changes, because every group of chat has own URL rule, but now we need to change it when language and dark-mode changes.